### ISSUE TITLE
Open terms can be type checked

### ISSUE THREAD

**Issue by francoisthire** [Deducteam/Dedukti#48]
The last version of master answers success on the following term and it should not since ?26, ?28, ?29 and ?30 are free variables. Actually, you can replace these variables by any term and it still works. On has to use the cic encoding to type check this term.

On Rodolphe implementation, this term is ill-typed.
```
eq :
  cic.Term (cic.type cic.z)
    (cic.prod (cic.type cic.z) (cic.type cic.z) (cic.univ cic.prop)
       (A:(cic.Univ cic.prop) =>
        cic.prod cic.prop (cic.type cic.z) A
          (_x:(cic.Term cic.prop A) =>
           cic.prod cic.prop (cic.type cic.z) A (__:(cic.Term cic.prop A) => cic.univ cic.prop)))).

refl :
  cic.Term cic.prop
    (cic.prod (cic.type cic.z) cic.prop (cic.univ cic.prop)
       (A:(cic.Univ cic.prop) =>
        cic.prod cic.prop cic.prop A (x:(cic.Term cic.prop A) => eq A x x))).

def match_eq_Prop :
  cic.Term cic.prop
    (cic.prod (cic.type cic.z) cic.prop (cic.univ cic.prop)
       (A:(cic.Univ cic.prop) =>
        cic.prod cic.prop cic.prop A
          (_x:(cic.Term cic.prop A) =>
           cic.prod (cic.type cic.z) cic.prop
             (cic.prod cic.prop (cic.type cic.z) A
                (__:(cic.Term cic.prop A) =>
                 cic.prod cic.prop (cic.type cic.z) (eq A _x __)
                   (z:(cic.Term cic.prop (eq A _x __)) => cic.univ cic.prop)))
             (return_type:
              (cic.Term ?26
                 (cic.prod cic.prop ?28 A
                    (__:(cic.Term ?30 A) =>
                     cic.prod cic.prop ?29 (eq A _x __)
                       (z:(cic.Term cic.prop (eq A _x __)) =>
                        cic.univ cic.prop)))) =>
              cic.prod cic.prop cic.prop (return_type _x (refl A _x))
                (case_refl:
                 (cic.Term cic.prop (return_type _x (refl A _x))) =>
                 cic.prod cic.prop cic.prop A
                   (__:(cic.Term cic.prop A) =>
                    cic.prod cic.prop cic.prop (eq A _x __)
                      (z:(cic.Term cic.prop (eq A _x __)) => return_type __ z))))))).
```

**Comment by GuillaumeGen** [comment]
It comes from commit #a6e6646

**Comment by francoisthire** [comment]
Fix by commit afe5c70
